ui/eslint.config.base.js

```diff
@@ -0,0 +1,35 @@
+const iTwinPlugin = require("@itwin/eslint-plugin");
+
+module.exports = [
+  {
+    files: ["**/*.{ts,tsx}"],
+    ...iTwinPlugin.configs.uiConfig,
+  },
+  {
+    files: ["**/*.{ts,tsx}"],
+    ...iTwinPlugin.configs.jsdocConfig,
+  },
+  {
+    files: ["**/*.{ts,tsx}"],
+    rules: {
+      "@typescript-eslint/ban-ts-comment": [
+        "error",
+        {
+          "ts-ignore": "allow-with-description",
+        },
+      ],
+      "@typescript-eslint/consistent-type-imports": "error",
+      "@typescript-eslint/no-unused-vars": [
+        "error",
+        {
+          args: "after-used",
+          argsIgnorePattern: "^_",
+          varsIgnorePattern: "^_",
+          ignoreRestSiblings: true,
+        },
+      ],
+      "no-duplicate-imports": "off",
+      "@itwin/no-internal-barrel-imports": "warn",
```

Comment:
This is a new addition I made, will come back to this after making this upgrade to presentation repo as well. Here to prevent lint failure for now.

Reply:
@aruniverse Should I be working on fixing the errors created by this. This seems to be newer (originating from newer eslint-plugin) and there are a ton of these ([build](https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=2268891&view=logs&jobId=9f3a3068-7880-5994-433d-b92c1f9d379b&j=9f3a3068-7880-5994-433d-b92c1f9d379b&t=83bcf1fc-5ebb-5f5c-0841-3e5c18e8878e)). I don't know if I should leave it to warn/off and have someone else on appui working on fixing it or if I should do it myself.

The lint errors seem to be coming from test files only (I think) and the meaning of error is this: 
![image](https://github.com/iTwin/appui/assets/98850418/b44e1679-cc23-4689-8ec3-ca796ce826cd)

Reply:
make a separate issue and have the appui team work on it, just suppress it for now

Reply:
Issue created: #413 , Should I assign to someone specific?